### ISSUE TITLE
issue on executing flutter doctor --android-licenses

### ISSUE THREAD

**Issue by akshat102** [flutter/flutter#71635]
this was my stack trace

Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
        at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
        at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
        at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
        ... 5 more

**Comment by darshankawar** [comment]
@akshat102 
Check out [this](https://github.com/flutter/flutter/issues/56778) issue and corresponding issue comments that has solutions to resolve this issue.
See if any of them resolves your issue.

**Comment by hello-addweb** [comment]
Please make sure you've Android SDK Tools installed in Android Studio, if you cant see it, unchecked hide obsolete packages and try `flutter doctor --android-licenses` again
can you please provide your `flutter doctor -v`?

For Android sdk tools got to : Tool -> SDK Manager

**Comment by jmewes** [comment]
I just got the same error on the attempt to run `flutter doctor --android-licenses`  on Ubuntu. In the linked issue it is recommend to use Java 8 to solve the issue. I can confirm that this works out. I just did it with the help of [SDKMAN!](https://sdkman.io/).

After this, the command worked out.

```
sdk list java
sdk install java 8.0.275.open-adpt
sdk use java 8.0.275.open-adpt
```

Afterwards I could run the command successfully.

In would be nice if Flutter would not depend upon an old JDK, though. Or if it would provide a straight forward error message like, "Expecting JDK 8. Found JDK 11".

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by ethael** [comment]
What additional information do you need to proceed @pedromassangocode? There is no lack of information, everything is here. All information were also in previous closed ticket #56778. 

#### problem 1
command `flutter doctor --android-licenses` relies on specific Java version and is not checking what version is locally installed. 

#### problem 2
it depends on old Java version. Current stable is version 11, in September this year there will be even newer LTS out (Java 17)

#### reason why it crashes on Java 11
because in Java 9 they removed java.xml.bind from core SDK to separate package. see [here](https://stackoverflow.com/questions/47155866/where-did-javax-xml-bind-go-in-java-9)

#### solution
add java.xml.bind as a manual dependency and not rely on having it directly in SDK
